packages/components/src/components/quickfact/quickfact.ts

```diff
@@ -0,0 +1,174 @@
+   * @internal
+   */
+  @state() defaultSlotIsFilled = false;
+
+  async show() {
+    this.open = true;
+    this.base.setAttribute('open', '');
+    return waitForEvent(this, 'sd-after-show');
+  }
+
+  async hide() {
+    this.open = false;
+    this.base.removeAttribute('open');
+    return waitForEvent(this, 'sd-after-hide');
+  }
+
+  @watch('open', { waitUntilFirstUpdate: true })
+  handleOpenChange() {
+    if (this.open) {
+      this.emit('sd-show');
+      this.show();
+      this.emit('sd-after-show');
+    } else {
+      this.emit('sd-hide');
+      this.hide();
+      this.emit('sd-after-hide');
+    }
+  }
+
+  render() {
+    return html`
+      <details
+        part="base"
+        class="flex items-center"
+        ?open=${this.open}
+        @toggle=${(e: ToggleEvent) => {
+          // Update the 'open' property based on the 'open' attribute of the <details> element
+          this.open = e.newState === 'open' ? true : false;
+        }}
+      >
+        <summary
+          part="summary-container"
+          class="flex flex-row sm:flex-col gap-4 mb-3 sm:mb-8 items-center text-center focus-visible:!focus-outline"
+        >
+          <slot name="icon"
+            ><sd-icon class="h-12 w-12 sm:h-24 sm:w-24" name="content/image" color="primary"></sd-icon
+          ></slot>
+
+          <slot
+            class=${cx(
+              'flex flex-col sm:gap-4 text-left sm:text-center',
+              this.defaultSlotIsFilled ? 'text-primary' : 'text-black'
+            )}
+            name="summary"
+          >
+          </slot>
+
+          <button
+            part="button"
+            class=${cx(
+              'ml-auto self-start sm:mx-auto text-primary transition-transform duration-300 ease-in-out focus-visible:!focus-outline',
+              !this.defaultSlotIsFilled && 'hidden',
+              this.open && 'rotate-180'
+            )}
+            @click=${() => {
+              this.open = !this.open;
+            }}
+          >
+            <div class=${cx(this.open && 'hidden')}>
+              <slot name="expand-icon">
+                <sd-icon
+                  class=${cx('h-6 w-6 sm:h-12 sm:w-12 grid place-items-center transition-transform')}
+                  library="system"
+                  name="chevron-down"
+                ></sd-icon>
+              </slot>
+            </div>
+
+            <div class=${cx(!this.open && 'hidden')}>
+              <slot name="collapse-icon">
+                <sd-icon
+                  class=${cx('h-6 w-6 sm:h-12 sm:w-12 grid place-items-center transition-transform')}
+                  library="system"
+                  name="chevron-down"
+                ></sd-icon>
+              </slot>
+            </div>
+          </button>
+        </summary>
+        <div part="content">
+          <slot
+            @slotchange=${() => {
+              this.defaultSlotIsFilled = this.defaultSlot.assignedElements().length > 0;
```

Comment:
As said this doesn't work for default slots without wrapper elements. If this is really needed, we either have to make the docs SUPER clear or would have to add a slot (unfortunately). In that case we would call it "details", to stay consistent with details/summary.